engine/src/main/java/org/hibernate/validator/internal/engine/ConstraintViolationImpl.java

```diff
@@ -196,7 +206,7 @@ public final Path getPropertyPath() {
 
 	@Override
 	public <C> C unwrap(Class<C> type) {
-		if ( type.isAssignableFrom( ConstraintViolation.class ) ) {
+		if ( type.isAssignableFrom( HibernateConstraintViolation.class ) ) {
```

Comment:
I guess instead of removing the original if statement, we should be able to just add `type.isAssignableFrom( HibernateConstraintViolation.class )` as a second if statement, right?